macOS/DuckDuckGo/Common/Database/CoreDataStore.swift

```diff
@@ -55,21 +56,7 @@ extension CoreDataStore {
 
 internal class CoreDataStore<ManagedObject: ValueRepresentableManagedObject> {
 
-    private let tableName: String
-    private var _readContext: NSManagedObjectContext??
-
-    private var readContext: NSManagedObjectContext? {
-        if case .none = _readContext {
-#if DEBUG
-            guard AppVersion.runType.requiresEnvironment else {
-                _readContext = .some(.none)
-                return .none
-            }
-#endif
-            _readContext = Database.shared.makeContext(concurrencyType: .privateQueueConcurrencyType, name: tableName)
```

Comment:
`CoreDataStore` is now independent on `Database.shared`. Instead, it takes an optional context or an optional database object that creates a context.

Because the database object is not passed to stores (DownloadsListStore, PermissionsStore, FireproofDomainsStore) and these stores belong to existing singletons, these singletons’ initialization is adjusted to  consider “requiresEnvironment” flag in Debug builds.